jfr-connection/src/test/java/io/opentelemetry/contrib/jfr/connection/RecordingConfigurationTest.java

```diff
@@ -0,0 +1,140 @@
+import java.util.Map;
+import javax.management.RuntimeMBeanException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.openjdk.jmc.common.item.IItem;
+import org.openjdk.jmc.common.item.IItemCollection;
+import org.openjdk.jmc.common.item.IItemIterable;
+import org.openjdk.jmc.flightrecorder.CouldNotLoadRecordingException;
+import org.openjdk.jmc.flightrecorder.JfrLoaderToolkit;
+
+public class RecordingConfigurationTest {
+
+  FlightRecorderConnection flightRecorderConnection = null;
+
+  @BeforeEach
+  public void setup() {
+    RecordingTest.deleteJfrFiles();
+    flightRecorderConnection = RecordingTest.getFlightRecorderConnection();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    RecordingTest.deleteJfrFiles();
+  }
+
+  @Test
+  public void nullConfigThrows() {
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> new RecordingConfiguration.JfcFileConfiguration(null));
+  }
+
+  @Test
+  public void brokenJfcConfigFileThrowsError() {
+    assertThrows(RuntimeMBeanException.class, () -> executeRecording("brokenJfcFile.jfc"));
+  }
+
+  @Test
+  public void jfcFileFromInputStreamCanBeRead() {
+    IItemCollection recordingContent = executeRecording("sampleJfcFile.jfc");
+    assertTrue(containsEvent(recordingContent, "jdk.ThreadAllocationStatistics"));
+  }
+
+  @Test
+  public void mapConfiguration() {
+
+    Map<String, String> recordingConfigAsMap = new HashMap<>();
+    recordingConfigAsMap.put("jdk.ObjectAllocationInNewTLAB#enabled", "true");
+    recordingConfigAsMap.put("jdk.ObjectAllocationOutsideTLAB#enabled", "true");
+
+    RecordingConfiguration recordingConfiguration =
+        new RecordingConfiguration.MapConfiguration(recordingConfigAsMap);
+
+    IItemCollection recordingContent = excecuteRecordingWithConfig(recordingConfiguration);
+    assertNotNull(recordingContent, "excecuteRecordingWithConfig returned null");
+    assertTrue(containsEvent(recordingContent, "jdk.ObjectAllocationInNewTLAB"));
+    assertTrue(containsEvent(recordingContent, "jdk.ObjectAllocationOutsideTLAB"));
+  }
+
+  private static boolean containsEvent(IItemCollection recordingContent, String eventName) {
+    for (IItemIterable iItemIterable : recordingContent) {
+      for (IItem iItem : iItemIterable) {
+        String currentEvent = iItem.getType().getIdentifier();
+        if (currentEvent.equals(eventName)) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  private IItemCollection executeRecording(String configFile) {
+    RecordingConfiguration.JfcFileConfiguration configuration =
+        new RecordingConfiguration.JfcFileConfiguration(
+            RecordingConfigurationTest.class.getClassLoader().getResourceAsStream(configFile));
+    return excecuteRecordingWithConfig(configuration);
+  }
+
+  private IItemCollection excecuteRecordingWithConfig(RecordingConfiguration configuration) {
```

Comment:
let this method throw `Exception` and remove all the catch/fail() chains inside.

Reply:
I'm okay with the verbosity as it helps document the code.